src/cmd/cmd.zig

```diff
@@ -618,23 +617,53 @@ fn getMyDataFromIpEcho(
 }
 
 fn getEntrypoints(logger: Logger) !std.ArrayList(SocketAddr) {
+    const EntrypointSet = std.ArrayHashMap(void, void, struct {
+        // zig fmt: off
+        pub fn hash(_: @This(), _: void) u32 { unreachable; }
+        pub fn eql(_: @This(), _: void, _: void, _: usize) bool { unreachable; }
+        // zig fmt: on
+    }, true);
+    const EntrypointCtx = struct {
+        entrypoints: []const SocketAddr,
+        pub fn hash(_: @This(), entrypoint: SocketAddr) u32 {
+            const array, const len = entrypoint.toString();
+            return std.array_hash_map.hashString(array[0..len]);
+        }
+        pub fn eql(ctx: @This(), a: SocketAddr, _: void, b_index: usize) bool {
+            return a.eql(&ctx.entrypoints[b_index]);
+        }
+    };
+
     var entrypoints = std.ArrayList(SocketAddr).init(gpa_allocator);
+    errdefer entrypoints.deinit();
+
+    var entrypoint_set = EntrypointSet.init(gpa_allocator);
+    defer entrypoint_set.deinit();
+
+    try entrypoint_set.ensureTotalCapacity(config.current.gossip.entrypoints.len);
+    try entrypoints.ensureTotalCapacityPrecise(config.current.gossip.entrypoints.len);
+
     for (config.current.gossip.entrypoints) |entrypoint| {
         const socket_addr = SocketAddr.parse(entrypoint) catch brk: {
-            // if we couldn't parse as IpV4, we attempt to resolve DNS and get IP
-            var domain_and_port = std.mem.splitScalar(u8, entrypoint, ':');
-            const domain_str = domain_and_port.next() orelse {
-                logger.field("entrypoint", entrypoint).err("entrypoint domain missing");
-                return error.EntrypointDomainMissing;
-            };
-            const port_str = domain_and_port.next() orelse {
+            const domain_port_sep = std.mem.indexOfScalar(u8, entrypoint, ':') orelse {
                 logger.field("entrypoint", entrypoint).err("entrypoint port missing");
                 return error.EntrypointPortMissing;
             };
+            const domain_str = entrypoint[0..domain_port_sep];
+            if (domain_str.len == 0) {
+                logger.field("entrypoint", entrypoint).err("entrypoint domain not valid");
```

Comment:
i know you didnt write this code but were avoiding using the field approach because of performance issues  - can you use fmt instead